src/Microsoft.DotNet.ImageBuilder/src/Commands/PublishManifestCommand.cs

```diff
@@ -77,7 +77,13 @@ await _registryCredentialsProvider.ExecuteWithCredentialsAsync(
                         .SelectMany(repo =>
                             repo.FilteredImages
                                 .Where(image => image.SharedTags.Any())
-                                .Select(image => (repo, image)));
+                                .Where(image => image.AllPlatforms
+                                    .Select(platform =>
+                                        ImageInfoHelper.GetMatchingPlatformData(platform, repo, imageArtifactDetails))
+                                    .Where(platformMapping => platformMapping.Platform != null)
+                                    .Any(t => !t.Platform!.IsUnchanged))
```

Comment:
This is another way to do this. I'm fine with either.

```suggestion
                                    .Any(t => t.Platform?.IsUnchanged == false))
```